restcomm/restcomm.identity/src/main/java/org/restcomm/connect/identity/EmailValidator.java

```diff
@@ -20,33 +20,33 @@
 
 package org.restcomm.connect.identity;
 
-import java.util.regex.Pattern;
+import javax.mail.internet.AddressException;
+import javax.mail.internet.InternetAddress;
+
+import org.apache.log4j.Logger;
 
 /**
  * @author ddh.huy@gmail.com (Huy Dang)
  */
 public class EmailValidator {
-    private static final String VALID_EMAIL_ADDRESS_REGEX = "^[_A-Za-z0-9-\\+]+(\\.[_A-Za-z0-9-]+)*@"
-                                                          + "[A-Za-z0-9-]+(\\.[A-Za-z0-9]+)*(\\.[A-Za-z]{2,})$";
+    private static Logger logger = Logger.getLogger(EmailValidator.class);
 
     /**
-     * Verify an email address has valid format or not, use the default regex
+     * Verify an email address has valid format or not
      *
      * @param email: the email address need to be verified
      * @return true if email format is valid, false if email format is invalid
      */
     public static boolean isValidEmailFormat ( String email ) {
-        return isValidEmailFormat(email, VALID_EMAIL_ADDRESS_REGEX);
-    }
-
-    /**
-     * Verify an email address has valid format or not, use the customized user input regex
-     *
-     * @param email: the email address need to be verified
-     * @return true if email format is valid, false if email format is invalid
-     */
-    public static boolean isValidEmailFormat ( String email, String regex ) {
-        Pattern pattern = Pattern.compile(regex);
-        return pattern.matcher(email).matches();
+        boolean isValid = false;
+        try {
+            InternetAddress emailAddress = new InternetAddress(email);
+            emailAddress.validate();
+            isValid = true;
+        } catch (AddressException ex) {
+            isValid = false;
+            logger.error("Email " + email + " is invalid");
```

Comment:
i would change the log level here to `warn` (if not `info`). 

(`error` is usually reserved for severe issues that affect the application health. on the other hand, this is very much a normal operation of handling invalid user input - `info` sounds like a better fit imho)

Reply:
@gsaslis Agree with you about the log level. I am prefer changing it to `warn`. Thanks for your suggestion.
About the [b001a77](https://github.com/RestComm/Restcomm-Connect/commit/b001a77c8ea7f9d1757315dd5affb2e5d52b157c), I see that we had two different classes with exactly same purpose. So I think it should be better if we only have one interface to raise the email invalid format exception.